dbt/adapters/redshift/connections.py

```diff
@@ -107,14 +135,115 @@ def get_connect_method(self):
+                    "Failed to use IdP credentials. 'region' must be provided."
+                )
+
+            if not self.credentials.password or not self.credentials.user:
+                raise dbt.exceptions.FailedToConnectError(
+                    "Failed to use IdP credentials. 'password' and 'user' must be provided."
+                )
+
+            if self.credentials.credentials_provider == "AzureCredentialsProvider":
+                if (
+                    not self.credentials.azure_idp_tenant
+                    or not self.credentials.azure_client_id
+                    or not self.credentials.azure_client_secret
+                    or not self.credentials.azure_preferred_role
+                ):
+                    raise dbt.exceptions.FailedToConnectError(
+                        "Failed to use Azure credential. 'azure_idp_tenant', 'azure_client_id', 'azure_client_secret', "
+                        "and 'azure_preferred_role' must be provided"
+                    )
+
+                def connect():
+                    logger.debug("Connecting to redshift with Azure Credentials Provider...")
+                    c = redshift_connector.connect(
+                        iam=True,
+                        region=self.credentials.region,
+                        database=self.credentials.database,
+                        cluster_identifier=self.credentials.cluster_id,
+                        credentials_provider="AzureCredentialsProvider",
+                        user=self.credentials.user,
+                        password=self.credentials.password,
+                        idp_tenant=self.credentials.azure_idp_tenant,
+                        client_id=self.credentials.azure_client_id,
+                        client_secret=self.credentials.azure_client_secret,
+                        preferred_role=self.credentials.azure_preferred_role,
+                    )
+                    return c
+
+                return connect
+            elif self.credentials.credentials_provider == "OktaCredentialsProvider":
+                if (
+                    not self.credentials.okta_idp_host
+                    or not self.credentials.okta_app_id
+                    or not self.credentials.okta_app_name
+                ):
+                    raise dbt.exceptions.FailedToConnectError(
+                        "Failed to use Okta credential. 'okta_idp_host', 'okta_app_id', 'okta_app_name' must be provided."
+                    )
+
+                def connect():
+                    logger.debug("Connecting to redshift with Okta Credentials Provider...")
+                    c = redshift_connector.connect(
+                        iam=True,
+                        region=self.credentials.region,
+                        database=self.credentials.database,
+                        cluster_identifier=self.credentials.cluster_id,
+                        credentials_provider="OktaCredentialsProvider",
```

Comment:
nit: does this need to be hardcoded? Looks like we already check that self.credentials.credentials_provider == "OktaCredentialsProvider" above

Reply:
Good call. Might also be a good idea to allow more wiggle room for the customer to use these credentials. I think we can hardcode credentials_provider="OktaCredentialsProvider" in the connection call, and allow for multiple variations of Okta from the customer(both "okta" and "oktacredentialprovider" should be accepted with case insensitivity), instead of requiring customers to input excatly "OktaCredentialsProvider"